api/v1alpha1/shared_types.go

```diff
@@ -275,3 +276,38 @@ const (
 	// https://github.com/google/re2/wiki/Syntax.
 	StringMatchRegularExpression StringMatchType = "RegularExpression"
 )
+
+// KubernetesHorizontalPodAutoscalerSpec defines Kubernetes Horizontal Pod Autoscaler settings of Envoy Proxy Deployment
+type KubernetesHorizontalPodAutoscalerSpec struct {
```

Comment:
I don't think this's the right direction, at last, you need borrow everything from `k8s.io.autoscaling.v2.HorizontalPodAutoScalerSpec`.

Reply:
if we still provide scaleTargetRef, then technically user could specify unrelated resource than Envoy Proxy Deployment

Reply:
Current approach looks well, any better ideas, IMO HPA fields are hard to abstract.